posts/2020-05-15-five-years-of-rust.md

```diff
@@ -0,0 +1,507 @@
+src/lib.rs:5:23: 5:24 error: cannot borrow `x` as immutable because it is also borrowed as mutable
+src/lib.rs:5     println!("{} {}", x, y);
+                                   ^
+note: in expansion of format_args!
+<std macros>:2:25: 2:56 note: expansion site
+<std macros>:1:1: 2:62 note: in expansion of print!
+<std macros>:3:1: 3:54 note: expansion site
+<std macros>:1:1: 3:58 note: in expansion of println!
+src/lib.rs:5:5: 5:29 note: expansion site
+src/lib.rs:3:18: 3:19 note: previous borrow of `x` occurs here; the mutable borrow prevents subsequent moves, borrows, or modification of `x` until the borrow ends
+src/lib.rs:3     let y = &mut x;
+                              ^
+src/lib.rs:6:2: 6:2 note: previous borrow ends here
+src/lib.rs:1 fn main() {
+src/lib.rs:2     let mut x = 7;
+src/lib.rs:3     let y = &mut x;
+src/lib.rs:4
+src/lib.rs:5     println!("{} {}", x, y);
+src/lib.rs:6 }
+             ^
+error: aborting due to previous error
+Could not compile `error-messages`.
+
+To learn more, run the command again with --verbose.
+```
+
+</details>
+
+![A terminal screenshot of the 1.2.0 error message.][borrow-error-1.2.0]
+
+<details>
+ <summary>1.43.0 Error Message</summary>
+
+```
+   Compiling error-messages v0.1.0 (/Users/ep/src/rust/error-messages)
+error[E0502]: cannot borrow `x` as immutable because it is also borrowed as mutable
+ --> src/lib.rs:5:23
+  |
+3 |     let y = &mut x;
+  |             ------ mutable borrow occurs here
+4 |
+5 |     println!("{} {}", x, y);
+  |                       ^  - mutable borrow later used here
+  |                       |
+  |                       immutable borrow occurs here
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0502`.
+error: could not compile `error-messages`.
+
+To learn more, run the command again with --verbose.
+```
+
+</details>
+
+![A terminal screenshot of the 1.43.0 error message.][borrow-error-1.43.0]
+
+[borrow-error-1.2.0]:  /images/2020-05-15-five-years-of-rust/borrow-error-1.2.0.png
+[borrow-error-1.43.0]: /images/2020-05-15-five-years-of-rust/borrow-error-1.43.0.png
+[help-error-1.2.0]:    /images/2020-05-15-five-years-of-rust/help-error-1.2.0.png
+[help-error-1.43.0]:   /images/2020-05-15-five-years-of-rust/help-error-1.43.0.png
+[trait-error-1.2.0]:   /images/2020-05-15-five-years-of-rust/trait-error-1.2.0.png
+[trait-error-1.43.0]:  /images/2020-05-15-five-years-of-rust/trait-error-1.43.0.png
+
+## Quotes from the teams
+Of course we can't cover every change that has happened. So we reached out and
+asked some of our teams what changes they are most proud of:
+
+> For rustdoc, the big things were:
+> * The automatically generated documentation for blanket implementations
+> * The search itself and its optimizations (last one being to convert it into JSON)
+> * The possibility to test more accurately doc code blocks "compile_fail,
+>   should_panic, allow_fail"
+> * Supporting blanket impls.
```

Comment:
this point is duplicated

Reply:
It's indeed duplicated with the first element of the list. I think we should replace it with the fact that the doc tests are now put into their own files.

Reply:
I've updated it now, let me know if it looks good.

Reply:
Thanks, it's better indeed. Just a little rewording though: "Doc tests can now be put into their own files." -> "Doc tests are now put into their own files."

Reply:
@GuillaumeGomez Can you explain what you mean by "Doc tests are now put into their own files."? I don't really follow what that does, I initially assumed it meant that you could put your doc tests in separate files but now I feel like that's not what this is?

Reply:
Sure, sorry, I should have precised: now we generate separate binaries. But I'm not sure it's worth talking about it...

Reply:
@GuillaumeGomez Oh no I think we should still include it if you want. I went with "Doc tests are now generated as their own seperate binaries." to clarify, does that work for you?

Reply:
Sounds good to me, thanks!